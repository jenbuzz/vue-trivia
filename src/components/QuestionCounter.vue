<template>
    <b-row align-h="center">
        <b-col sm="6">
            <p class="text-center mb-3">{{ currentCount }} / {{ totalCount }}</p>
        </b-col>
    </b-row>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component
export default class QuestionCounter extends Vue {
    @Prop() private readonly currentCount!: number;
    @Prop() private readonly totalCount!: number;
}
</script>
