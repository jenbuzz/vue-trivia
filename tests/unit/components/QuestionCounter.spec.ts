import { createLocalVue, shallowMount } from '@vue/test-utils';
import BootstrapVue from 'bootstrap-vue';
import QuestionCounter from '@/components/QuestionCounter.vue';

const localVue = createLocalVue();
localVue.use(BootstrapVue);

describe('Component: QuestionCounter', () => {
    it('renders', () => {
        const wrapper = shallowMount(QuestionCounter, {
            localVue,
            propsData: { currentCount: 1, totalCount: 2 },
        });

        expect(wrapper).toMatchSnapshot();
    });
});
