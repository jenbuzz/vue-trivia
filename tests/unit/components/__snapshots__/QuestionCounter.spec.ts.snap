// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: QuestionCounter renders 1`] = `
<b-row-stub tag="div" alignh="center">
  <b-col-stub sm="6" tag="div">
    <p class="text-center mb-3">1 / 2</p>
  </b-col-stub>
</b-row-stub>
`;
